rosidl_runtime_py/convert.py

```diff
@@ -174,6 +183,11 @@ def message_to_ordereddict(
     """
     d = OrderedDict()
 
+    # Special case: if the message is serialized, then return a single field containing the bytes
+    if isinstance(msg, bytes):
+        d['bytes'] = msg
+        return d
```

Comment:
This seems to be a pretty awkward API. Why should this function handle this case? It sounds like the caller could check this and don't even call the function - especially since the special returned dict must be handled differently already.

Reply:
>  Why should this function handle this case?

My thinking was that a serialized ROS message is still a ROS message. I thought it was better to write the logic handling serialized versus deserialized messages once in a common place instead of guarding against passing serialized messages to this function. I'll admit the number of places where this may happen is likely small, so I'm also okay with adding guards wherever it can happen.

> It sounds like the caller could check this and don't even call the function - especially since the special returned dict must be handled differently already.

Where this is used currently, we don't need to handle the returned dict differently (https://github.com/ros2/ros2cli/pull/470); it still results in a valid dict and YAML representation.

Reply:
> I thought it was better to write the logic handling serialized versus deserialized messages once in a common place instead of guarding against passing serialized messages to this function.

Don't you already have to special case this in the calling code? Since the message doesn't have a `bytes` field (which is what the returned dictionary expresses).

> Where this is used currently, we don't need to handle the returned dict differently

I would argue that this should be handled (and printed) differently since otherwise it isn't distinguishable from a message with a `bytes` field. The output should clarify that this is a serialized message.

Reply:
Makes sense; We don't want to confuse a with a field named `bytes`. :+1: for clarify the output is a serialized message. I can make related changes to ros2topic instead of relying on this PR.